scripts/synthetix/facilicate_trade.js

```diff
@@ -0,0 +1,97 @@
+const { getExchange } = require("../utils/trading_strategy_helpers")(web3, artifacts)
+const { SynthetixJs } = require("synthetix-js")
+const { calculateBuyAndSellAmountsFromPrice, fetchTokenInfoFromExchange } = require("../utils/trading_strategy_helpers")(web3, artifacts)
+
+// TODO - put this somewhere more generic. Copied from dex-contracts
+const sendTxAndGetReturnValue = async function (method, ...args) {
+  const result = await method.call(...args)
+  await method(...args)
+  return result
+}
+
+// truffle uses network by name and SynthetixJS uses network by ID
+const networkMap = {
+  "mainnet": 1,
+  "rinkeby": 4,
+}
+
+const argv = require("yargs")
+  .option("spread", {
+    type: "float",
+    describe: "Percentage increase required for trade (fees not accounted)",
+    default: 0.2,
+  })
+  .version(false).argv
+
+module.exports = async callback => {
+  try {
+    const snxjs = new SynthetixJs({ networkId: networkMap[argv.network] })
+
+    const exchange = await getExchange(web3)
+    const account = (await web3.eth.getAccounts())[0]
+
+    // Both of these hardcoded tokens have 18 decimal places.
+    // We "trust" that this will always be the case
+    // Although it seems that synthetix has the authority to upgade their token
+    // which could mean deploying a new one with a different number of decimal places.
+    const sETHKey = await snxjs.sETH.currencyKey()
+    const sUSDKey = await snxjs.sUSD.currencyKey()
+
+    const sETHAddress = await snxjs.Synthetix.synths(sETHKey)
+    const sUSDAddress = await snxjs.Synthetix.synths(sUSDKey)
```

Comment:
I'm wondering if we shouldn't use the addresses of the tokens that are listed on dƒusion as if they upgraded the addresses we would have to first list the new tokens anyways in order to place orders.

Reply:
Ya, I'd be happy to do that. Makes for less contract calls!

Reply:
Only problem I have with this then would be that I need to keep a mapping of the token addresses for at least both `rinkeby` and `mainnet` networks. I'll check it out.

Reply:
Done anyway.